### ISSUE TITLE
Logging error on Windows (ValueError: underlying buffer has been detached)

### ISSUE THREAD

**Issue by RUrlus** [scikit-build/scikit-build#651]
Hi, thanks for the new release! Unfortunately I am getting a logging error on the Windows GitHub runners, no issues on MacOS and Linux. Any idea what could be triggering this?

```
--- Logging error ---
Traceback (most recent call last):
File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\logging\__init__.py", line 1088, in emit
stream.write(msg + self.terminator)
ValueError: underlying buffer has been detached
Call stack:
...
File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-51dm91dw\overlay\Lib\site-packages\setuptools\command\install_lib.py", line 12, in run
outfiles = self.install()
File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-51dm91dw\overlay\Lib\site-packages\skbuild\command\install_lib.py", line 18, in install
distutils_log.info("copied %d files" % len(outfiles))
Message: 'copied 29 files'
Arguments: ()
```

See https://github.com/KaveIO/PhiK/runs/5049900568?check_suite_focus=true#step:5:447 for the full trace

**Comment by henryiii** [comment]
That's weird. Setuptools uses `logging` now (v 60.2 or 60, something like that). I'd probably be okay to just ignore the ValueError, as it's just a logging failure, but would be nice to solve. My guess is setuptools is detaching the logger - we don't mess with it.

**Comment by henryiii** [comment]
I added an experimental fix. Could you try that? Is that the only time (during install) that the logger is detached?

Also, `setuptools<60` would probably also be a temporary fix, but I wouldn't recommend it generally.

**Comment by RUrlus** [comment]
Hi @henryiii, thanks for the quick response.

>  Is that the only time (during install) that the logger is detached?

It happens multiple times but always on `distutils_log.info("copied %d files" % self.outfiles_count)` from `skbuild\command\build_py.py` at line 41.

See the full stack-trace here: https://gist.github.com/RUrlus/cf4c70201f12ad40b7350e7f9c416494.

It also seems we're not the only one, I ran across a similar issue [here](https://github.com/maxbachmann/RapidFuzz/blob/6b79b02f1e997339401239fbb471475cc9cfee87/.github/workflows/releasebuild.yml#L12)

**Comment by henryiii** [comment]
My worry is that if we fix this one, the other logger statements might then be reachable, and also be broken. Could you test the patch?

**Comment by henryiii** [comment]
I've forked and triggered the workflow here: https://github.com/henryiii/PhiK/runs/5075313527?check_suite_focus=true - we'll see.

**Comment by henryiii** [comment]
Looks like that fixes it! :D

**Comment by RUrlus** [comment]
> Looks like that fixes it! :D

@henryiii, strange but the runs failed on Windows for Phik when I point to the commit (c18845a1c8a0af9fd9f04a531b501a17650bfa98) using `"scikit-build @ git+https://github.com/scikit-build/scikit-build@c18845a1c8a0af9fd9f04a531b501a17650bfa98"
`

**Comment by henryiii** [comment]
Is this `NotImplementedError: Patefield requires a compiled extension that was not found.`? Strange, I didn't get that.

**Comment by henryiii** [comment]
Ahh, but I only ran the cibuildwheel job. Maybe something is missing here?

**Comment by RUrlus** [comment]
> Is this `NotImplementedError: Patefield requires a compiled extension that was not found.`?

This is expected as the build using skbuild fails but other build without extension has no issue. The compiled extension is more a nice-to-have than strictly necessary for Phik. Hence, we try installing with the extension using skbuild's setup, we catch the exception and try to install it without the extension. For the secondary install we use setuptools setup.

The exception you're see above is caused by a test calling a function that relies on the extension, this means that the skbuild setup failed but the setuptools one did not.

It still fails on the same error in the build phase, it's just not a hard exception

**Comment by henryiii** [comment]
Can you disable the fallback in https://github.com/KaveIO/PhiK/pull/44/files so I can see what the error is? I would not be at all surprised if you are hitting #650 and that's fixed now (and same recommendation goes here, you should make ninja conditional on not being on Windows, MSVC comes with Ninja and currently it's better at finding it's own MSVC with it).

**Comment by henryiii** [comment]
FYI, 0.13.1 is out.

**Comment by RUrlus** [comment]
@henryiii resolved by 0.13.1 and the suggested changes, thanks!

For reference the stack-trace of the logging error now being caught.

```removing _skbuild\win-amd64-3.8\setuptools\bdist.win-amd64\wheel
--- Logging error ---
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\logging\__init__.py", line 1088, in emit
    stream.write(msg + self.terminator)
ValueError: underlying buffer has been detached
Call stack:
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 363, in <module>
    main()
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 345, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 261, in build_wheel
    return _build_backend().build_wheel(wheel_directory, config_settings,
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\build_meta.py", line 229, in build_wheel
    return self._build_with_temp_dir(['bdist_wheel'], '.whl',
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\build_meta.py", line 214, in _build_with_temp_dir
    self.run_setup()
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\build_meta.py", line 157, in run_setup
    exec(compile(code, __file__, 'exec'), locals())
  File "setup.py", line 152, in <module>
    sk_setup(**setup_args, **sk_build_kwargs)
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\skbuild\setuptools_wrap.py", line 699, in setup
    return setuptools.setup(*args, **kw)
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\__init__.py", line 155, in setup
    return distutils.core.setup(**attrs)
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 148, in setup
    return run_commands(dist)
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 163, in run_commands
    dist.run_commands()
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 967, in run_commands
    self.run_command(cmd)
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 986, in run_command
    cmd_obj.run()
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\skbuild\command\bdist_wheel.py", line 38, in run
    super(bdist_wheel, self).run(*args, **kwargs)
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\wheel\bdist_wheel.py", line 299, in run
    self.run_command('build')
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\_distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 986, in run_command
    cmd_obj.run()
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\_distutils\command\build.py", line 135, in run
    self.run_command(cmd_name)
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\_distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 986, in run_command
    cmd_obj.run()
  File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-c7gsbccr\overlay\Lib\site-packages\skbuild\command\build_py.py", line 41, in run
    distutils_log.info("copied %d files" % self.outfiles_count)
Message: 'copied 19 files'
Arguments: ()
```
